src/components/admin/store/PickupOrder/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { Button, Typography } from '@/components/common';
+import { OrderStatusIndicator } from '@/components/store';
+import { StoreAPI } from '@/lib/api';
+import { UUID } from '@/lib/types';
+import { PublicOrderWithItems } from '@/lib/types/apiResponses';
+import { OrderStatus } from '@/lib/types/enums';
+import { getOrderItemQuantities, itemToString, reportError } from '@/lib/utils';
+import { useState } from 'react';
+import styles from './style.module.scss';
+
+interface PickupOrderProps {
+  token: string;
+  canFulfill: boolean;
+  order: PublicOrderWithItems;
+  onOrderUpdate: (orders: PublicOrderWithItems) => void;
+}
+
+const PickupOrder = ({ token, canFulfill, order, onOrderUpdate }: PickupOrderProps) => {
+  const itemQuantities = getOrderItemQuantities(order.items);
+  const [unselected, setUnselected] = useState(new Set<UUID>());
+
+  return (
+    <tr className={styles.row}>
+      <td>
+        <Typography variant="h5/regular">{`${order.user.firstName} ${order.user.lastName}`}</Typography>
+        <OrderStatusIndicator orderStatus={order.status} />
+        {canFulfill ? (
+          <Button
+            size="small"
+            onClick={async () => {
+              try {
+                const items = order.items.filter(item => !unselected.has(item.uuid));
+                const newOrder = await StoreAPI.fulfillOrderPickup(token, order.uuid, items);
+                const itemUuids = items.map(item => item.uuid);
+                onOrderUpdate({
+                  ...newOrder,
+                  items: order.items.map(item =>
+                    itemUuids.includes(item.uuid) ? { ...item, fulfilled: true } : item
+                  ),
+                });
```

Comment:
seeking to learn: does the API response not just return the whole new order updated correctly? do we need to manually set fulfilled for the items we fulfilled

Reply:
the API response doesn't return the `items` array so it won't say which items were fulfilled